packages/react-dom/src/__tests__/ReactDOMHooks-test.js

```diff
@@ -105,6 +109,172 @@ describe('ReactDOMHooks', () => {
     expect(labelRef.current.innerHTML).toBe('abc');
   });
 
+  it('useUniqueID does not change id even if the component updates during client render', async () => {
+    const {act} = TestUtils;
+    const {useUniqueID, useState} = React;
+    let _setShowId;
+    function App() {
+      const id = useUniqueID();
+      const [showId, setShowId] = useState(false);
+      _setShowId = setShowId;
+      return (
+        <div>
+          <div aria-labelledby={id} />
+          {showId && <div id={id} />}
+        </div>
+      );
+    }
+
+    const root = ReactDOM.createRoot(container);
+    await act(async () => {
+      root.render(<App />);
+    });
+
+    expect(container.innerHTML).toMatchInlineSnapshot(
```

Comment:
if you do `expect(container).toMatchInlineSnapshot` `pretty-format` will format the inner html as html (newlines and less escaping, closer to how you'd write html manually).

If you explicitly want `innerHTML`, you can do `expect(wrap(container.innerHTML)).toMatchInlineSnapshot` to at least avoid the inline escaping and wrapping quotes. `wrap` comes from `jest-snapshot-serializer-raw` and is already used in `react-refresh`'s tests